vignettes/projpred.Rmd

```diff
@@ -0,0 +1,320 @@
+```{r, fig.asp=1.5 * 0.618}
+plot(cvvs, stats = c("elpd", "rmse"), deltas = TRUE)
+```
+
+Based on that plot, we would decide for a model size of 8 because that's the point where the performance measures level off and are close enough to the reference model's performance.
+```{r}
+modsize_decided <- 8
+```
+
+Note that **projpred** offers the `suggest_size()` function which may help in the decision for a model size, but this is a rather heuristic method and needs to be interpreted with caution (see the `?suggest_size` help).
+```{r}
+suggest_size(cvvs)
+```
+Here, we would get the same final model size (`8`) as by our manual decision.
+
+Only now, after we have made a decision for the model size, we inspect further results from the variable selection and, in particular, the solution path. For example, we can simply `print()` the resulting object:
+```{r}
+cvvs
+### Alternative modifying the number of printed decimal places:
+# print(cvvs, digits = 2)
+### 
+```
+The solution path can be seen in the `print()` output (column `solution_terms`), but it is also accessible through the `solution_terms()` function:
+```{r}
+( soltrms <- solution_terms(cvvs) )
+```
+
+Combining the decided model size of 8 with the solution path leads to the following terms (as well as the intercept) as the predictor terms of the final submodel:
+```{r}
+( soltrms_final <- head(soltrms, modsize_decided) )
+```
+
+## Post-selection inference
+
+The `project()` function returns an object of class `projection` which forms the basis for convenient post-selection inference. By the following code, `project()` will project the reference model onto the final submodel once again^[During the forward search, the reference model has already been projected onto all candidate models (this was where arguments `ndraws` and `nclusters` of `cv_varsel()` came into play). During the evaluation of the submodels along the solution path, the reference model has already been projected onto those submodels (this was where arguments `ndraws_pred` and `nclusters_pred` of `cv_varsel()` came into play). In principle, one could use the results from the evaluation part for post-selection inference, but due to a bug in the current implementation (see GitHub issue #168), we currently have to project once again.]:
+```{r, cache=TRUE, cache.lazy=FALSE}
+prj <- project(
+  refm_fit,
+  solution_terms = soltrms_final,
+  seed = 15705533
+)
+```
+<!-- Alternative, as soon as GitHub issue #168 is resolved: -->
+<!-- ```{r} -->
+<!-- prj <- project( -->
+<!--   cvvs, -->
+<!--   nterms = modsize_decided, -->
+<!--   cv_search = FALSE, -->
+<!--   seed = 15705533 -->
+<!-- ) -->
+<!-- ``` -->
+For more accurate results, we could have increased argument `ndraws` of `project()` (up to the number of posterior draws in the reference model). This increases the runtime, though, which we don't want in this vignette.
```

Comment:
"This increases the runtime, though, which we don't want in this vignette."
->
This increases the runtime, which we don't want in this vignette.